src/http_client.ts

```diff
@@ -23,13 +17,25 @@ const defaultOptions = {
     apiBaseUrl: "https://api.logflare.app",
 }
 
+class NetworkError extends Error {
+    name = "NetworkError"
+
+    constructor(
+        message: string,
+        public response: Response,
+        public data: unknown
+    ) {
+        super(message)
+    }
+}
```

Comment:
This NetworkError class is a little helper that is similar to the type of errors that Axios throws, and allows us to add the response and any data, so they can be used in a catch block. You can see it used on line 102 below.